[EDD] WooTube 3주차 기능구현

kimjungmin-developer opened the pull request:
안녕하세요! 이번주 구현 기능 목록입니다.

* 구독 CRUD 기능 구현
* 댓글 CRUD 기능 구현
* 채널페이지 API 연동
* 조회수 API
* 회원/로그인 관련 API

kimjungmin-developer committed:
[feat] get current user api (#151)

* 현재 로그인한 사용자 정보를 가져오는 api입니다.

Laterality committed:
[feat] add view count to video and change to retrieve with it

Laterality committed:
[feat] impolement update view count

Laterality committed:
[fix] fix issue with adding xss prevention to video publish request (#150)

비디오 게시 요청의 인자들이 XSS escaping을 거치도록 수정합니다.

Laterality committed:
[feat] update video lsit and detali view to show view count

Laterality committed:
[feat] update db schema for production to add view count column

Laterality committed:
[fix] fix issue (#158)

- 중복 작성된 라인을 지웠습니다.

heebong committed:
[feat] add navi subcribe area

heebong committed:
[refactor] rename method

heebong committed:
[refactor] rename method common.js

heebong committed:
[refactor] change return type

Laterality committed:
[feat] 사용자 정보 수정 및 탈퇴 API 연동 (#162)

* 웹페이지에서 사용자 정보 수정과 탈퇴가 동작하도록 합니다.
* app.css 파일을 교체합니다.

heebong committed:
[refactor] add console.log

kangmin46 committed:
[feat] 내 채널 페이지에서의 비디오 조회 (#163)

* 채널 페이지에서 해당 사용자가 올린 비디오를 조회할 수 있도록 합니다.

heebong committed:
[refactor] add fetch - catch

heebong committed:
[refactor] change error response

heebong committed:
[refactor] change response

heebong committed:
[refactor] change err response

heebong committed:
[refactor] change err response message

Laterality committed:
[refactor] update test to be based callback

Laterality committed:
[fix] fix issue with request saving in order

heebong committed:
[refactor] change err response message

heebong committed:
Merge branch 'develop' of https://github.com/everyone-driven-development/miniprojects-2019 into feature/front-subscribe

# Conflicts:
#	frontend/js/common.js

heebong committed:
[fix] resolve merge conflict

heebong committed:
[refactor] change comment

Laterality committed:
[style] adjust method location

Laterality committed:
[feat] apply peer feedback

Laterality committed:
[feat] remove unnecessary line

Laterality committed:
[refactor] rename service method

Laterality committed:
[refactor] add column annotation to video entity

Laterality committed:
조회수 갱신 및 조회 (#159)

* 비디오에 조회수 값을 추가합니다.
* 비디오를 id로 조회할 때마다 조회수를 증가시켜 반환하도록 합니다.

heebong committed:
[refactor] change url address

Laterality committed:
Merge pull request #164 from heebong/feature/front-subscribe

* 좌측에 있던 사이드바 네비게이션의 중복 제거
* common.js에 현재 로그인된 상태인지 확인하는 isLogin 함수 추가
* common.js에 구독을 가지고오는 requestSubcribed 함수 추가
* 로그인이 된 상황이면 requestSubcribed 함수를 실행해 구독한 사람들을 사이드바 네비게이션에 추가
* 사이드바 네비게이션의 라이브러리 탭은 클릭시 미 로그인시엔 로그인 페이지로, 아닐 땐 본인 채널 페이지로 이동하게 변경

Laterality committed:
Merge pull request #171 from Laterality/issue/166

리스트 조회를 위한 동영상 게시 요청을 순차적으로 호출하도록 변경합니다.

Laterality committed:
[feat] add logback setting for file logging in production environment (#173)

- 프로덕션 환경에서 파일에 로그를 기록하는 설정을 추가합니다.

kangmin46 committed:
[feat] CommentCRUD (#174)

* 코멘트의 생성 , 조회, 수정, 삭제 기능 구현
* 일부 테스트 코드 리팩토링

Conatuseus committed:
[feat]add  connection logout Api (#176)

- 로그아웃 API 연동
- 헤더의 로그인 확인 기능 ( 로그인 한 상황 : 상단 헤더에 로그아웃, 로그인 하지 않은 상황 : 상단 헤더에 로그인 표시 )

kangmin46 committed:
[fix] issue 168 (#177)

* 중국말 한국어로 패치, 코드 포매팅

kimjungmin-developer committed:
[feat] add front comment

- 댓글 작성/수정/삭제 마크업 및 ajax 통신 기능을 추가했습니다.

kimjungmin-developer committed:
[feat] add subscription (#180)

*  구독 API 구독 추가했습니다

Laterality committed:
fix #182 (#185)

* 서버 응답의 날짜 포맷을 변경합니다.
* 이에 따른 프론트 페이지의 날짜 표기와 관련된 코드를 수정합니다.

heebong committed:
[fix] front index no header  (#186) (#188)

* 헤더와 사이드 바가 나오지 않던 문제를 해결했습니다.

kimjungmin-developer committed:
[refactor] change front comment delete btn (#193)

* 댓글 삭제, 수정 버튼 클릭을 더 수월하게 바꿨습니다.

heebong committed:
[refactor] add subscription schema (#194)

subscription schema를 추가했습니다.

kangmin46 committed:
[feat] 나의 채널 페이지에서 나의 동영상들 조회 API (#196)

- 나의 채널 페이지에서 나의 동영상 조회하도록 API 수정

* [feat] add requestMyChannelVideos api in common.js,

* [refactor] change baseUrl

* [chore] delete conflict trace

* [refactor] delete duplicate common.js method

* [refactor] delete unnecessary blank

* [feat] add findbyCrator

* [feat] add front user create video

* [refactor] remove empty line

* [refactor] remove empty line

Laterality committed:
fix #197 (#198)

* [fix] change source of bootstrap js/css

* [fix] change jquery cdn to jsdelivr

* [refactor] remove loading script integrity attribute

* [refactor] remove integrity attribute

kimjungmin-developer committed:
[fix] video user delete bug (#199)

- 탈퇴한 유저가 작성한 비디오를 가져올 때 발생하는 오류를 해결했습니다.

kangmin46 committed:
[fix] Issue/201 (#202)

* 같은 이메일로 가입시 익셉션
* 이메일, 이름, 비밀번호 null시 익셉션

Laterality committed:
[fix] fix to check whether if password matches its confirmation (#204)

* 회원가입시 비밀번호 확인도 함께 확인하도록 변경합니다.

heebong committed:
[fix] password check fail tst (#210)

*컴파일 에러 이슈 해결

heebong committed:
[fix] password confirm password (#212)

* 비밀번호 확인 validation 체크시 프론트에서 요청을 해주지 않았던 이슈 해결

Laterality committed:
[refactor] change front video detail page style (#211)

- 비디오 상세 페이지 본문과 댓글 영역 스타일을 조정합니다.

Laterality committed:
[feat] change to show videos in view count order (#213)

- 동영상 조회수 순으로 노출하는 기능 추가했습니다.

kimjungmin-developer committed:
[feat] feature/subscription button (#214)

* 프론트엔드 기능 구현완료

heebong committed:
[refactor] change front user leave btn issue (#208) (#215)

- 프론트에서 회원 탈퇴 버튼 클릭시 무조건 로그인한 사용자가 탈퇴되는 현상을 해결했습니다.

heebong committed:
[refactor] add email duplicate error (#217)

탈퇴한 이메일로 회원가입 시도시 DB 에러 캐치했습니다.

wmakerjun commented on frontend/js/common.js:
```diff
@@ -1,29 +1,19 @@
+        getUrlParams: function (name, url) {
+            if (!url) url = window.location.href;
+            name = name.replace(/[\[\]]/g, '\\$&');
+            var regex = new RegExp('[?&]' + name + '(=([^&#]*)|&|#|$)'),
```
var 대신에 let 또는 const를 작성해주세요 :)

Conatuseus replied:
const 로 변경했습니다 💯

wmakerjun commented on frontend/js/common.js:
```diff
@@ -1,29 +1,19 @@
+            return localResponseDate.fromNow();
         }
     },
     constants : {
```
4 line과 같이 :에 대한 공백 컨벤션을 통일시켜주세요 :)

Conatuseus replied:
변경했습니다.

wmakerjun commented on frontend/js/video/comment.js:
```diff
@@ -0,0 +1,166 @@
+const commentCtx = {
+    state: {
+        currentCommentId: null,
+    },
```
오류를 발생 시킬 수도 있는 콤마 입니다. 뒤에 연결해서 쓸 값이 없다면 콤마를 지워 주세요 :)

Conatuseus replied:
필요없는 콤마 제거 했습니다!! 👍

wmakerjun commented on frontend/js/headerTemplate.js:
```diff
@@ -1,21 +1,29 @@
+
+    const signedInHeaderTemplate =
+        `<header id="header" class="header navbar shadow-sm">
         <div class="header-container">
```
공백 컨벤션을 js 파일안에서 맞춰주면 더 좋습니다 :)

Conatuseus replied:
공백 컨벤션 맞춰서 수정했습니다 🥇

wmakerjun commented on frontend/js/headerTemplate.js:
```diff
@@ -31,31 +39,30 @@ const addHeader = function (headerElement){
+
+    const notSignedInHeaderTemplate =
+        `<header id="header" class="header navbar shadow-sm">
         <div class="header-container">
```
공백 컨벤션을 js 파일안에서 맞춰주면 더 좋습니다 :)

wmakerjun commented on frontend/js/signup.js:
```diff
@@ -2,11 +2,13 @@ const handleSignUpEvent = function () {
     const nameElm = document.querySelector('input[type=name]')
     const emailElm = document.querySelector('input[type=email]')
     const passwordElm = document.querySelector('input[type=password]')
+    const passwordComElm = document.querySelector('input[name=password_confirm]')
```
html 페이지에서 id, class, 그리고 다른 속성명들도 *-* 형태로 선언하고 있으니 password-confirm으로 통일시키면 어떨까요~?

wmakerjun commented on frontend/js/video/comment.js:
```diff
@@ -0,0 +1,166 @@
+
+        const prepareEdit = function(event) {
+            const { target } = event
+            if(target.classList.contains('btn-update-comment') ||
```
if 다음에 괄호 사이에 [공백컨벤션](https://ui.toast.com/fe-guide/ko_CODING-CONVENSION/#%EA%B3%B5%EB%B0%B1)에 맞게 공백을 넣어주세요 :)

wmakerjun commented on frontend/js/video/comment.js:
```diff
@@ -0,0 +1,166 @@
+
+        const deleteComment = function(event) {
+            const { target } = event;
+            if(target.classList.contains('btn-delete-comment') ||
```
if 다음에 괄호 사이에 공백컨벤션에 맞게 공백을 넣어주세요 :)

wmakerjun commented on frontend/video-detail.html:
```diff
@@ -2,6 +2,7 @@
 <html>
 
 <head>
+        <meta charset="utf-8">
```
head안에 다른 태그들과 시작하는 위치를 맞춰주세요 :) tab이 한번 더 들어간 것 같아요

wmakerjun reviewed (changes requested):
피드백 남길게 별로 없을 정도로 너무너무너무 잘하고 있습니다 💯💯💯💯💯
몇가지 사소한 컨벤션과 작은 실수들에 대해서만 피드백 남겼으니 리팩토링할 때 참고해주세요 😄

kangmin46 committed:
[fix] resolve issue 228 (#230)

- comment, user XSS 추가했습니다.

kimjungmin-developer committed:
[fix] fix subscription bug (#232)

- 구독관련 함수를 common.js에 추가했습니다.

kimjungmin-developer committed:
[fix] fix duplicatEmailSignUpException (#233)

- 이메일 중복 메세지 내용을 변경했습니다.

heebong committed:
[fix] video authenticate (#220) (#234)

- 비디오 수정/삭제시 현재 로그인 사용자를 확인하는 로직을 추가했습니다.

Laterality committed:
[feature] fix cors issue (#235)

- CORS 관련 필터를 추가하여 비로그인 상태에서 로그인이 필요한 요청 시 발생하는 401 응답을 기존 응답과 동일하게 처리할 수 있도록 수정합니다.
- 프론트엔드의 일부 불필요한 코드를 정리합니다.

Laterality committed:
[fix] update video comment count #221 (#236)

- 비디오 상세 페이지에서 댓글 작성과 삭제마다 댓글 개수를 갱신하도록 합니다.

Laterality committed:
[fix] change front video detail comment area markup  #222 (#237)

- 댓글의 수정/삭제 버튼이 작성자에게만 보이도록 수정합니다.

kangmin46 committed:
[fix] #226 회원가입 비디오 게시 회원 수정 관련 size 검증 (#238)

* 회원가입 수정 게시 관련 size 검증

Conatuseus committed:
[fix] change email validation #229 (#239)

email validation 추가.
기존 : ~~ @ ~~~
변경: ~~ @ ~~~ . ~~~ 가 되도록.
변경에 따른 테스트 코드 추가

Conatuseus committed:
[fix] fix message for name (#231)

- 닉네임이 영어, 한글이 가능하다고 변경

Laterality committed:
[fix] fix #223 (#242)

동영상 상세 페이지의 동영상 수정/삭제 버튼이 작성자에게만 보이도록 변경합니다.

heebong committed:
[refactor] change UserSaveRequestDto Name message (#245)

회원가입시 이름 에러 메세지를 변경했습니다.

Laterality committed:
[fix] change front channel page (#247)

- 비로그인상태거나 자신 이외의 사용자가 채널 페이지에 접근하는 경우 회원정보 수정과 회원 탈퇴 버튼을 노출하지 않도록 변경합니다.

gracefulBrown commented on backend/build.gradle:
```diff
@@ -31,6 +31,7 @@ dependencies {
     implementation 'org.springframework.boot:spring-boot-starter-data-jpa'
     implementation 'org.springframework.boot:spring-boot-starter-web'
     implementation 'org.springframework.boot:spring-boot-starter-webflux'
+    implementation 'com.navercorp.lucy:lucy-xss:1.6.3'
```
xss 필터를 적용하였군요 :)

Laterality committed:
[fix] add xss to update dto  (#249)

- domain을 수정할 때도 xss 방어 로직하도록 수정했습니다.

gracefulBrown commented on backend/src/main/java/com/woowacourse/edd/application/service/SubscriptionInternalService.java:
```diff
@@ -0,0 +1,67 @@
+        checkSelfSubscription(subscribedId, userId);
+
+        User subscribed = userInternalService.findById(subscribedId);
+        User subscriber = userInternalService.findById(userId);
```
한번 고민해보면 좋을 것 같아서 남기는데요~
각각 두번의 디비 접근 대신 한번의 접근으로 해결해보는거는 어떨까요?

gracefulBrown commented on backend/src/test/java/com/woowacourse/edd/presentation/controller/VideoControllerTests.java:
```diff
@@ -101,22 +155,31 @@ void save_invalid_contents() {
+        Long id = Long.valueOf(urls[urls.length - 1]);
         String youtubeId = "updateYoutubeId";
         String title = "updateTitle";
         String contetns = "updateContents";
```
오타가 있네요 :)

gracefulBrown commented on backend/src/test/java/com/woowacourse/edd/presentation/controller/VideoControllerTests.java:
```diff
@@ -128,61 +191,107 @@ void update_invalid_id() {
+            .exchange();
+    }
+
+    private WebTestClient.BodyContentSpec saveNextVideo(VideoSaveRequestDto video, String sid) {
```
사용하는 메서드 인가요?

heebong committed:
[refactor] change front subscription area(#244) (#250)

- 구독 신청/취소 시 사이드 네비의 구독목록 리스트에 바로 반영되게 변경했습니다.
- 사이드 네비의 라이브러리 부분의 이름을 로그인 했으면 본인 이름으로 노출되게 변경했습니다

gracefulBrown commented on backend/src/main/java/com/woowacourse/edd/presentation/controller/LoginController.java:
```diff
@@ -15,6 +18,7 @@
 
     public static final String LOGIN_URL = "/v1/login";
```
@ApiVersion 사용 검토를 해보면 좋을 것 같네요 :)

gracefulBrown commented on backend/src/main/java/com/woowacourse/edd/presentation/controller/SubscriptionController.java:
```diff
@@ -0,0 +1,58 @@
+        this.subscriptionService = subscriptionService;
+    }
+
+    @PostMapping("/{subscribedId}/subscribe")
```
subscribedId가 userId가 맞나요?
api path가 v1/users/{subscribedId}/subscribe 로 되어있을 때 subscribedId가 user의 Id 임을 좀 더 나타내주는 변수명을 사용하면 좋을 것 같아요 :) (예를들면 subscriberId라 던지..ㅎㅎ;;)

gracefulBrown commented on backend/src/main/java/com/woowacourse/edd/presentation/controller/SubscriptionController.java:
```diff
@@ -0,0 +1,58 @@
+        return ResponseEntity.ok(subscriptionCountResponse);
+    }
+
+    @GetMapping("/{id}/subscribed")
```
다른 path는 subscribedId를 사용하는데 해당 API는 id로 사용하고 있네요? 컨벤션 지켜주세요 ^^

gracefulBrown reviewed (approved):
메서드 분리 등 코드 정리가 깔끔하게 잘 되어있네요 👍 
간단한 피드백 몇가지 남겼습니다 :)
참고해서 프로젝트 잘 마무리 해주세요 ^^

heebong committed:
[fix] change mypage video (#253)

- 채널페이지에서 비디오 리스트를 가져올 수 있도록 변경했습니다.

Conatuseus committed:
[refactor] feedback (#248)

준 피드백 반영

kimjungmin-developer committed:
[feedback]feedback/252 (#254)

kang-hyungu commented on backend/src/main/java/com/woowacourse/edd/application/converter/CommentConverter.java:
```diff
@@ -0,0 +1,25 @@
+    }
+
+    public static CommentRequestDto escapeUpdateRequestDto(CommentRequestDto updateRequestDto) {
+        return new CommentRequestDto(XssPreventer.escape(updateRequestDto.getContents()));
```
스프링에 필터나 인터셉터를 추가해서 쿼리스트링, 리퀘스트 바디에서 한 번 거르면 모든 converter 클래스에 XssPreventer를 쓰지 않아도 될 것 같네요

kang-hyungu commented on backend/src/main/java/com/woowacourse/edd/application/service/SubscriptionInternalService.java:
```diff
@@ -0,0 +1,78 @@
+import java.util.List;
+
+@Service
+public class SubscriptionInternalService {
```
트랜잭션 어노테이션이 누락된 것 같네요

kang-hyungu commented on backend/src/main/java/com/woowacourse/edd/config/CORSFilter.java:
```diff
@@ -0,0 +1,32 @@
+public class CORSFilter implements Filter {
+
+    @Override
+    public void doFilter(ServletRequest servletRequest, ServletResponse servletResponse, FilterChain filterChain)
```
필터에서 htttp 헤더에 직접 작성하지 말고 스프링 부트에서 CORS 지원하는 기능을 써보면 어떨까요

kang-hyungu reviewed (approved):
피드백 늦어서 죄송해요!
몇 가지 간단한 피드백 남겼어요
보시고 궁금한 점 있으면 DM 주시고, 프로젝트 마무리 잘 하시기 바랍니다! :)

heebong committed:
[refactor] change front main page video count (#258)

woowahan-pjs reviewed (approved):
축하합니다! 백엔드 Java 레벨 2의 모든 미션 완료하였어요. 🥇 
나만의 블로그부터 미니 프로젝트까지
수많은 미션을 수행하느라 대단히 고생 많으셨습니다. 💯 
좋은 개발자가 되기 위한 소중한 밑거름이 되었길 바랍니다.
이만 Merge 하겠습니다.
